### PR TITLE
CU-263dx6e - Mosaic assets mapping

### PR DESCRIPTION
**Pull request by hussein-aitlahcen**
- implements `LocalToRemoteAsset` and `RemoteToLocalAsset` to map a `(network_id, remote_asset_id) -> local_asset_id` and `(network_id, local_asset_id) -> remote_asset_id`
- refactor extrinsics to allow the relayer to specify `(network_id, remote_asset_id)` only (remote_asset_id highly likely to be a ETH token address)
- refactor tests & benchmarking to fit the changes

### CODE REVIEW

**Comment by ComposableFiB0t** [comment]
Task linked: [CU-263dx6e Pallet-Mosaic asset mapping](https://app.clickup.com/t/263dx6e)

**Comment by hussein-aitlahcen** [comment]
NOTE: I did remove duplicated tests.

**Review by benluelo — APPROVED** [review]

File: runtime/dali/src/lib.rs [hunk 1]
```diff
@@ -910,6 +910,7 @@ impl mosaic::Config for Runtime {
 	type MinimumTimeLockPeriod = MinimumTimeLockPeriod;
 	type BudgetPenaltyDecayer = mosaic::BudgetPenaltyDecayer<Balance, BlockNumber>;
 	type NetworkId = u32;
+	type RemoteAssetId = [u8; 20];
```
Comment (KaiserKarel) [inline runtime/dali/src/lib.rs:913]: Can we make this an enum with 1 variant for now, as we will be storing these, we may want to add others later on. Enums will be easier to migrate storage wise.

**Review by KaiserKarel — DISMISSED** [review]

**Review by KaiserKarel — DISMISSED** [review]

**Review by andor0 — APPROVED** [review]

**Comment by hussein-aitlahcen** [comment]
Ongoing fix #684

**Review by seunlanlege — APPROVED** [review]

**Pull request merged**